centreon/src/Core/HostCategory/Infrastructure/API/DeleteHostCategory/DeleteHostCategoryController.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+/*
+* Copyright 2005 - 2022 Centreon (https://www.centreon.com/)
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*
+* For more information : contact@centreon.com
+*
+*/
+
+declare(strict_types=1);
+
+namespace Core\HostCategory\Infrastructure\API\DeleteHostCategory;
+
+use Centreon\Application\Controller\AbstractController;
+use Core\HostCategory\Application\UseCase\DeleteHostCategory\DeleteHostCategory;
+use Core\Infrastructure\Common\Api\DefaultPresenter;
+
+final class DeleteHostCategoryController extends AbstractController
+{
+    public function __invoke(
+        int $hostCategoryId,
+        DeleteHostCategory $useCase,
+        DefaultPresenter $presenter
+    ): object {
```

Comment:
Return type hinting more precise ? `Response`

Reply:
As existing controller seems to return object as well, I left it like that for now